File: C# Basics/Console-In-and-Out-Homework/QuadraticEquation/QuadraticEquation.cs
using System;
using System.Globalization;
using System.Threading;
/* Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots). */
class Program
{
    static void Main()
    {
        Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;

        double a = double.Parse(Console.ReadLine());
        double b = double.Parse(Console.ReadLine());
        double c = double.Parse(Console.ReadLine());

        double d = Math.Pow(b, 2) - (4 * a * c);
        if (d > 0)
        {
            double firstRoot = (-b - Math.Sqrt(d)) / (2 * a);
            double secondRoot = (-b + Math.Sqrt(d)) / (2 * a);
            if (firstRoot > secondRoot)
            {
                firstRoot += secondRoot;
                secondRoot = firstRoot - secondRoot;
                firstRoot = firstRoot - secondRoot;
            }

            Console.WriteLine("{0:0.00}", firstRoot);
            Console.WriteLine("{0:0.00}", secondRoot);
        }
        else if (d == 0)
        {
            double root = -b / (2 * a);
            Console.WriteLine("{0:0.00}", root);
        }
        else
        {
            Console.WriteLine("no real roots");
        }
    }
}

File: JavaScript Basics/Conditional statements - Homework/quadratic-equation.js
function solve(args) {
    var a = +args[0],
        b = +args[1],
        c = +args[2],
        D,
        x1,
        x2;

    D = b * b - 4 * a * c;
    if (D > 0) {
        x1 = (-b - Math.sqrt(D)) / (2 * a);
        x2 = (-b + Math.sqrt(D)) / (2 * a);
        return 'x1=' + x1.toFixed(2) + '; x2=' + x2.toFixed(2);
    } else if (D === 0) {
        x1 = -b / (2 * a);
        x2 = x1;
        return 'x1=x2=' + x1.toFixed(2);
    } else {
        return 'no real roots';
    }
}

console.log(solve([2, 5, -3]));
console.log(solve([-1, 3, 0]));
console.log(solve([-0.5, 4, -8]));
console.log(solve([5, 2, 8]));
console.log(solve([0.2, 9.572, 0.2]));
